pkg/background/update_request_controller.go

```diff
@@ -298,11 +265,71 @@ func (c *Controller) deleteUR(obj interface{}) {
 			return
 		}
 	}
+	// sync Handler will remove it from the queue
+	c.enqueueUpdateRequest(ur)
+}
 
-	if ur.Status.Handler != "" {
-		return
+func (c *Controller) processUR(ur *kyvernov1beta1.UpdateRequest) error {
+	switch ur.Spec.Type {
+	case kyvernov1beta1.Mutate:
+		ctrl, _ := mutate.NewMutateExistingController(c.kyvernoClient, c.client,
+			c.policyLister, c.npolicyLister, c.urLister, c.eventGen, logger, c.configuration)
+		return ctrl.ProcessUR(ur)
+
+	case kyvernov1beta1.Generate:
+		ctrl, _ := generate.NewGenerateController(c.kyvernoClient, c.client,
+			c.policyLister, c.npolicyLister, c.urLister, c.eventGen, c.nsLister, logger, c.configuration,
+		)
+		return ctrl.ProcessUR(ur)
 	}
+	return nil
+}
 
-	// sync Handler will remove it from the queue
-	c.enqueueUpdateRequest(ur)
+func (c *Controller) markUR(name string) (*kyvernov1beta1.UpdateRequest, bool, error) {
+	var ok bool
+	var ur *kyvernov1beta1.UpdateRequest
+	err := retry.RetryOnConflict(common.DefaultRetry, func() error {
+		var err error
+		ur, err = c.urLister.Get(name)
+		if err != nil {
+			return err
+		}
+		ur = ur.DeepCopy()
+		if ur.Status.Handler != "" {
+			ok = ur.Status.Handler == config.KyvernoPodName()
+			return nil
+		}
+		ur.Status.Handler = config.KyvernoPodName()
+		ur, err = c.kyvernoClient.KyvernoV1beta1().UpdateRequests(config.KyvernoNamespace()).UpdateStatus(context.TODO(), ur, metav1.UpdateOptions{})
+		return err
+	})
+	return ur, ok, err
```

Comment:
Should we return a deepcopy-ed UR as it is used later during update?

Reply:
No need, already deepcopy-ed line 300